Community/Tdarr_Plugin_Soggys_NVENC_HEVC_CQV_Optimised_Bitrate.js

```diff
@@ -0,0 +1,220 @@
+    preset: '',
+    container: '.mkv',
+    handBrakeMode: false,
+    FFmpegMode: false,
+    reQueueAfter: true,
+    infoLog: '',
+  }
+
+  const streamHeight = file.ffProbeData.streams[0].height;
+  const streamWidth = file.ffProbeData.streams[0].width;
+  const streamFPS = file.mediaInfo.track[0].FrameRate;
+  let streamBR = file.bit_rate;
+
+  //check if the file is a video, if not the function will be stopped immediately
+  if (file.fileMedium !== 'video') {
+    response.processFile = false
+    response.infoLog += '☒File is not a video! \n'
+    return response
+  } else {
+    // bitrateprobe = file.ffProbeData.streams[0].bit_rate;
+    response.infoLog += '☑File is a video! \n'
+  }
+
+  optimalVideoBitrate = Math.floor((streamHeight * streamWidth * streamFPS) * inputs.targetCodecCompression);
+  response.infoLog += 'Optimal Bitrate = ' + (optimalVideoBitrate / 1000000).toString() + '\n'
+  response.infoLog += 'Stream Bitrate = ' + (streamBR / 1000000).toString() + '\n'
+  if (file.ffProbeData.streams[0].codec_name == 'hevc'
+  && (isNaN(optimalVideoBitrate) || isNaN(streamBR) || streamBR < optimalVideoBitrate)) {
+    response.processFile = false
+    response.infoLog += '☑File is already in hevc and is below optimal bitrate!\n'
+    return response
+  }
+  else if (file.ffProbeData.streams[0].codec_name == 'hevc'
+  && (file.video_resolution === '480p' || file.video_resolution === '576p' || file.video_resolution === '720p')) {
+    response.processFile = false
+    response.infoLog += '☑File is low res, wont transcode!\n'
+    return response
+  }
+
+  // Check if preset is configured, default to slow if not
+  var ffmpeg_preset
+  if (inputs.ffmpeg_preset === undefined) {
+    ffmpeg_preset = `slow`
+    response.infoLog += '☑Preset not set, defaulting to slow\n'
+  } else {
+    ffmpeg_preset = `${inputs.ffmpeg_preset}`
+    response.infoLog += `☑Preset set as ${inputs.ffmpeg_preset}\n`
+  }
+
+  //codec will be checked so it can be transcoded correctly
+  if (file.video_codec_name == 'h263') {
+    response.preset = `-c:v h263_cuvid`
+  } else if (file.video_codec_name == 'h264') {
+    if (file.ffProbeData.streams[0].profile != 'High 10') {
+      //Remove HW Decoding for High 10 Profile
+      response.preset = `-c:v h264_cuvid`
+    }
+  } else if (file.video_codec_name == 'mjpeg') {
+    response.preset = `-c:v mjpeg_cuvid`
+  } else if (file.video_codec_name == 'mpeg1') {
+    response.preset = `-c:v mpeg1_cuvid`
+  } else if (file.video_codec_name == 'mpeg2') {
+    response.preset = `-c:v mpeg2_cuvid`
+  } else if (file.video_codec_name == 'vc1') {
+    response.preset = `-c:v vc1_cuvid`
+  } else if (file.video_codec_name == 'vp8') {
+    response.preset = `-c:v vp8_cuvid`
+  } else if (file.video_codec_name == 'vp9') {
+    response.preset = `-c:v vp9_cuvid`
+  } else if (file.video_codec_name == 'hevc') {
+    if (file.ffProbeData.streams[0].profile != 'High 10') {
+      response.preset = `-c:v hevc_cuvid`
+    }
+  }
```

Comment:
I don't know the exact details, but this can also be replaced with `-hwaccel cuda` instead. This makes it so the GPU/ffmpeg decides what to do.

If you decide to keep this, make it a switch statement and use strict checking. `===`

Reply:
that worked! thanks